### DISCUSSION TITLE
X-Postal-Msgid Header and other Privacy-prone headers...

### DISCUSSION THREAD

**Discussion by krull** [postalserver/postal#2661] [Feature suggestions]
Hello Postal!

Thank you for a wonderful platform! :tada:

I wanted to contribute something back to the community..

In retrospect to https://github.com/postalserver/postal/issues/86, I wanted to inquire the use-case of  `X-Postal-Msgid` header, because I noticed that headers exposes you are using Postal as your sending platform.

Additionally, I noticed that when using the API, your initial Received header also states 'by Postal with HTTP', unlike [here](https://github.com/postalserver/postal/blob/2f0155747c5c058afb83856ce5c3c8d5f8cad341/lib/postal/smtp_server/client.rb#L319) in this code.

I managed to change both `.rb` file [here](https://github.com/postalserver/postal/blob/2f62baa238fc1102706ee4acf079b7a876b05283/app/models/incoming_message_prototype.rb#L94) and [here](https://github.com/postalserver/postal/blob/2f62baa238fc1102706ee4acf079b7a876b05283/app/models/outgoing_message_prototype.rb#L178) to use the config entry `smtp_server_hostname`, and it worked.

So my question is: other than a few hard-coded places in the code-base when searching through the github repo, are there any reasoning to use `X-Postal-Msgid` header as-is? Is there any way we can have it use an entry in the postal.yml file? What is it's intended purpose in life? :thinking: 

Thank you all in advance, 

@krull

**Comment by willpower232** [comment]
Presumably you can use it for cross referencing with the API if you're emailing yourself but yeah, not much use for the most part.

As for exposing that you're using Postal, that depends entirely on your threat model and how exposed you would feel if other people knew about your use of Postal.

If you want to make the changes and make a PR then I'm sure someone would find it useful at some point.
